tests/cstor/invalidconfig/invalidconfig_test.go

```diff
@@ -0,0 +1,231 @@
+	"github.com/openebs/maya/tests/cstor"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("[cstor] [-ve] TEST INVALID STORAGEPOOLCLAIM", func() {
+	var (
+		err error
+	)
+
+	AfterEach(func() {
+		By("deleting resources created for testing cstor volume provisioning", func() {
+
+			By("listing spc")
+			spcList, err = ops.SPCClient.List(metav1.ListOptions{})
+			Expect(err).To(BeNil(), "while listing spc clients", spcList)
+
+			By("deleting spc")
+			for _, spc := range spcList.Items {
+				_, err = ops.SPCClient.Delete(spc.Name, &metav1.DeleteOptions{})
+				Expect(err).To(BeNil(), "while deleting the spc's", spc)
+			}
+			time.Sleep(5 * time.Second)
+		})
+	})
+
+	When("creating storagepoolclaim with invalid disk type", func() {
+		It("should not create any cstorpool", func() {
+
+			By("building spc object")
+			spcObj = spc.NewBuilder().
+				WithName(spcName).
+				WithDiskType("invalid-disk-type").
+				WithMaxPool(cstor.PoolCount).
+				WithOverProvisioning(false).
+				WithPoolType(string(apis.PoolTypeStripedCPV)).
+				Build().Object
+
+			By("creating storagepoolclaim")
+			_, err = ops.SPCClient.Create(spcObj)
+			Expect(err).To(BeNil(), "while creating spc", spcName)
+
+			By("verifying healthy csp count as 0")
+			cspCount := ops.GetHealthyCSPCount(spcName, cstor.PoolCount)
+			Expect(cspCount).To(Equal(0), "while checking cstorpool health status")
+
+		})
+	})
+
+	When("creating storagepoolclaim with invalid pool type", func() {
+		It("should not create any cstorpool", func() {
+
+			By("building spc object")
+			spcObj = spc.NewBuilder().
+				WithName(spcName).
+				WithDiskType(string(apis.TypeSparseCPV)).
+				WithMaxPool(cstor.PoolCount).
+				WithOverProvisioning(false).
+				WithPoolType(string("invalid-pool-type")).
+				Build().Object
+
+			By("creating storagepoolclaim")
+			_, err = ops.SPCClient.Create(spcObj)
+			Expect(err).To(BeNil(), "while creating spc", spcName)
+
+			By("verifying healthy csp count as 0")
+			cspCount := ops.GetHealthyCSPCount(spcName, cstor.PoolCount)
+			Expect(cspCount).To(Equal(0), "while checking cstorpool health status")
+
+		})
+	})
+
+	When("creating storagepoolclaim with invalid pool count", func() {
+		It("should not create any cstorpool", func() {
+
+			By("building spc object")
+			spcObj = spc.NewBuilder().
+				WithName(spcName).
+				WithDiskType(string(apis.TypeSparseCPV)).
+				WithMaxPool(-1).
+				WithOverProvisioning(false).
+				WithPoolType(string(apis.PoolTypeStripedCPV)).
+				Build().Object
+
+			By("creating storagepoolclaim")
+			_, err = ops.SPCClient.Create(spcObj)
+			Expect(err).To(BeNil(), "while creating spc", spcName)
+
+			By("verifying healthy csp count as 0")
+			cspCount := ops.GetHealthyCSPCount(spcName, cstor.PoolCount)
+			Expect(cspCount).To(Equal(0), "while checking cstorpool health status")
+
+		})
+	})
+
+})
+
+var _ = Describe("[cstor] TEST INVALID STORAGECLASS", func() {
```

Comment:
Let move this to a new file e.g. `tests/cstor/pool/negative/invalid_storageclass_test.go`